src/app/store/slices/storage/storage.thunks/uploadItemsThunk.ts

```diff
@@ -604,7 +605,7 @@ export const uploadItemsParallelThunkNoCheck = createAsyncThunk<void, UploadItem
     const filesToUpload: FileToUpload[] = [];
     const errors: Error[] = [];
     const tasksIds: string[] = [];
-    const abortController = new AbortController();
+    const abortController = options?.abortController || new AbortController();
```

Comment:
This allows to control this operation from another operation, the problem with the event driven cancellation is that if we are doing parallel operations, some of them can get in between the event emit and the actual operation cancel, is not a problem usually since the time is in ms, but for multiple operations it causes problems like this case, so now we can abort this from the parent directly if we pass this option

Reply:
Yeah, passing the abort controller is the best way. The issue here is that the abort controller, apart from using the signal to ensure any task that is going on if you want, should follow a procedural pattern where a reason is passed to the `abort` function as this will replicate any other error. Take into count that if you cancel a fetch via passing the signal, that fetch is going to fail, at least this is the recommended way to follow. We can go with this anyway.